### PR TITLE
Improve nvgGlobalCompositeOperation with state

### PR DESCRIPTION
**Pull request by satoren**
Currently nvgGlobalCompositeOperation not supported with nvgSave and nvgRestore.

### CODE REVIEW

File: example/demo.c [hunk 1]
```diff
@@ -1060,6 +1066,58 @@ void drawScissor(NVGcontext* vg, float x, float y, float t)
...
+        NVG_SOURCE_IN,
+        NVG_SOURCE_OUT,
+        NVG_ATOP,
+        NVG_DESTINATION_OVER,
+        NVG_DESTINATION_IN,
+        NVG_DESTINATION_OUT,
+        NVG_DESTINATION_ATOP,
+        NVG_LIGHTER,
+        NVG_COPY,
+        NVG_XOR};
```
Comment (memononen) [inline example/demo.c]: curlie on new line

File: example/demo.c [hunk 2]
```diff
@@ -1060,6 +1066,58 @@ void drawScissor(NVGcontext* vg, float x, float y, float t)
...
+    
+    int count = sizeof(ops)/sizeof(ops[0]);
+
+
```
Comment (memononen) [inline example/demo.c]: only one empty line

File: example/demo.c [hunk 3]
```diff
@@ -1060,6 +1066,58 @@ void drawScissor(NVGcontext* vg, float x, float y, float t)
...
+        NVG_DESTINATION_OUT,
+        NVG_DESTINATION_ATOP,
+        NVG_LIGHTER,
+        NVG_COPY,
+        NVG_XOR};
+    
+    int count = sizeof(ops)/sizeof(ops[0]);
+
+    
+    for(i=0;i<count;++i)
```
Comment (memononen) [inline example/demo.c]: be consistent with formatting
```C
for (i = 0; i < count; i++) {
```

File: src/nanovg_gl.h [hunk 4]
```diff
@@ -1099,19 +1125,24 @@ static GLenum glnvg_convertBlendFuncFactor(int factor)
...
+	blend.dstRGB = glnvg_convertBlendFuncFactor(op.dstRGB);
+	blend.srcAlpha = glnvg_convertBlendFuncFactor(op.srcAlpha);
+	blend.dstAlpha = glnvg_convertBlendFuncFactor(op.dstAlpha);
+	if (blend.srcRGB == GL_INVALID_ENUM || blend.dstRGB == GL_INVALID_ENUM || blend.srcAlpha == GL_INVALID_ENUM || blend.dstAlpha == GL_INVALID_ENUM)
+	{
+		blend.srcRGB = GL_ONE;
+		blend.dstRGB = GL_ONE_MINUS_SRC_ALPHA;
+		blend.srcAlpha = GL_ONE;
+		blend.dstAlpha = GL_ONE_MINUS_SRC_ALPHA;
+    }
```
Comment (memononen) [inline src/nanovg_gl.h]: tab

File: example/demo.c [hunk 5]
```diff
@@ -1060,6 +1066,58 @@ void drawScissor(NVGcontext* vg, float x, float y, float t)
...
+    {
+        nvgSave(vg);
+        
+        nvgTranslate(vg, x + i * 40, y);
+        
+        nvgGlobalCompositeOperation(vg, NVG_COPY);
```
Comment (memononen) [inline example/demo.c]: You could just draw a circle and rectangle, no need for images.

Comment (satoren) [inline example/demo.c]: Need pre-render image. 
just like this.
http://www.html5canvastutorials.com/advanced/html5-canvas-global-composite-operations-tutorial/

Comment (memononen) [inline example/demo.c]: Can you elaborate why? This should work equally well:
```C
nvgBeginPath(vg);
nvgRect(vg, 0, 0, 20, 20);
nvgFillColor(vg, nvgRGBA(0,0,255,255));
nvgFill(vg);

nvgBeginPath(vg);
nvgCircle(vg, 20, 20, 10);
nvgFillColor(vg, nvgRGBA(255,0,0,255));
nvgFill(vg);
```

Comment (satoren) [inline example/demo.c]: Demo code is removed.

Comment (memononen) [inline example/demo.c]: I think we should keep it. But we should try to figure out why the results are different with images and shapes. They should be the same.

Comment (satoren) [inline example/demo.c]: I can't think of an efficient implementation to the same specification as HTML5 for nvgGlobalCompositeOperation.

Comment (satoren) [inline example/demo.c]: Thinking you do not merge if this problem is not resolved? I think this PR is not direct related to the problem.

**Comment by AndrewBelt** [comment]
I'm using this for my project and I haven't had any problems. Consider merging?

**Comment by memononen** [comment]
@AndrewBelt thanks for reporting that. Merged!

**Pull request merged**